mrp_workcenter_account_move/model/mrp.py

```diff
@@ -38,10 +38,11 @@ def test_accounting_setting(self):
         self.ensure_one()
         msg = ''
         if not self.routing_id:
-            msg_journal = _('Please set a Journal in BoM: {bom} to book '
-                            'Production Cost Journal Entries\n')
-            if not self.bom_id.journal_id:
-                msg += msg_journal.format(bom=self.bom_id.name)
+            msg_journal = _(
+                'Please set a Journal in Product Category: {cat} to book '
+                'Production Cost Journal Entries\n')
+            if not self.product_id.categ_id.property_stock_journal:
+                msg += msg_journal.format(cat=self.product_id.categ_id.name)
             if msg:
```

Comment:
There no need for any of that,

We are just getting rid of a field which and instead we are using a previous existing one.

provision exist to avoid people from booking Journal Entries without that field